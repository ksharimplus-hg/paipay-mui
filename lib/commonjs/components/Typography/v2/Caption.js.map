{"version":3,"names":["Caption","props","styles","text","style","StyleSheet","create","fontSize","lineHeight","marginVertical","letterSpacing"],"sources":["Caption.tsx"],"sourcesContent":["import * as React from 'react';\nimport { Text, TextStyle, StyleSheet, StyleProp } from 'react-native';\n\nimport StyledText from './StyledText';\n\nexport type Props = React.ComponentProps<typeof Text> & {\n\tstyle?: StyleProp<TextStyle>;\n\tchildren: React.ReactNode;\n};\n\n// @component-group Typography\n\n/**\n * Typography component for showing a caption.\n *\n * <div class=\"screenshots\">\n *   <img src=\"screenshots/caption.png\" />\n * </div>\n *\n * ## Usage\n * ```js\n * import * as React from 'react';\n * import { Caption } from 'react-native-paper';\n *\n * const MyComponent = () => (\n *   <Caption>Caption</Caption>\n * );\n *\n * export default MyComponent;\n * ```\n */\nconst Caption = (props: Props) => (\n\t<StyledText\n\t\t{...props}\n\t\talpha={0.54}\n\t\tfamily=\"regular\"\n\t\tstyle={[styles.text, props.style]}\n\t/>\n);\n\nexport default Caption;\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\tfontSize: 12,\n\t\tlineHeight: 20,\n\t\tmarginVertical: 2,\n\t\tletterSpacing: 0.4,\n\t},\n});\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAAsC;AAAA;AAAA;AAAA;AAOtC;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,MAAMA,OAAO,GAAIC,KAAY,iBAC5B,oBAAC,mBAAU,eACNA,KAAK;EACT,KAAK,EAAE,IAAK;EACZ,MAAM,EAAC,SAAS;EAChB,KAAK,EAAE,CAACC,MAAM,CAACC,IAAI,EAAEF,KAAK,CAACG,KAAK;AAAE,GAEnC;AAAC,eAEaJ,OAAO;AAAA;AAEtB,MAAME,MAAM,GAAGG,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,QAAQ,EAAE,EAAE;IACZC,UAAU,EAAE,EAAE;IACdC,cAAc,EAAE,CAAC;IACjBC,aAAa,EAAE;EAChB;AACD,CAAC,CAAC"}