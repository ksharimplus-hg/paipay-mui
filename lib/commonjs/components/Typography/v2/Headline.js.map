{"version":3,"names":["Headline","props","styles","text","style","StyleSheet","create","fontSize","lineHeight","marginVertical","letterSpacing"],"sources":["Headline.tsx"],"sourcesContent":["import * as React from 'react';\nimport { Text, TextStyle, StyleSheet, StyleProp } from 'react-native';\n\nimport StyledText from './StyledText';\n\nexport type Props = React.ComponentProps<typeof Text> & {\n\tstyle?: StyleProp<TextStyle>;\n\tchildren: React.ReactNode;\n};\n\n// @component-group Typography\n\n/**\n * Typography component for showing a headline.\n *\n * <div class=\"screenshots\">\n *   <img src=\"screenshots/headline.png\" />\n * </div>\n *\n * ## Usage\n * ```js\n * import * as React from 'react';\n * import { Headline } from 'react-native-paper';\n *\n * const MyComponent = () => (\n *   <Headline>Headline</Headline>\n * );\n *\n * export default MyComponent;\n * ```\n */\nconst Headline = (props: Props) => {\n\treturn (\n\t\t<StyledText\n\t\t\t{...props}\n\t\t\talpha={0.87}\n\t\t\tfamily=\"regular\"\n\t\t\tstyle={[styles.text, props.style]}\n\t\t/>\n\t);\n};\n\nexport default Headline;\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\tfontSize: 24,\n\t\tlineHeight: 32,\n\t\tmarginVertical: 2,\n\t\tletterSpacing: 0,\n\t},\n});\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAAsC;AAAA;AAAA;AAAA;AAOtC;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,MAAMA,QAAQ,GAAIC,KAAY,IAAK;EAClC,oBACC,oBAAC,mBAAU,eACNA,KAAK;IACT,KAAK,EAAE,IAAK;IACZ,MAAM,EAAC,SAAS;IAChB,KAAK,EAAE,CAACC,MAAM,CAACC,IAAI,EAAEF,KAAK,CAACG,KAAK;EAAE,GACjC;AAEJ,CAAC;AAAC,eAEaJ,QAAQ;AAAA;AAEvB,MAAME,MAAM,GAAGG,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,QAAQ,EAAE,EAAE;IACZC,UAAU,EAAE,EAAE;IACdC,cAAc,EAAE,CAAC;IACjBC,aAAa,EAAE;EAChB;AACD,CAAC,CAAC"}