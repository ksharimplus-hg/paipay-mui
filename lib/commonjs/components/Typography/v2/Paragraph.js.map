{"version":3,"names":["Paragraph","props","styles","text","style","StyleSheet","create","fontSize","lineHeight","marginVertical","letterSpacing"],"sources":["Paragraph.tsx"],"sourcesContent":["import * as React from 'react';\nimport { TextProps, StyleSheet } from 'react-native';\n\nimport StyledText from './StyledText';\n\nexport type Props = TextProps & {\n\tchildren: React.ReactNode;\n};\n\n// @component-group Typography\n\n/**\n * Typography component for showing a paragraph.\n *\n * <div class=\"screenshots\">\n *   <img src=\"screenshots/paragraph.png\" />\n * </div>\n *\n * ## Usage\n * ```js\n * import * as React from 'react';\n * import { Paragraph } from 'react-native-paper';\n *\n * const MyComponent = () => (\n *   <Paragraph>Paragraph</Paragraph>\n * );\n *\n * export default MyComponent;\n * ```\n */\nconst Paragraph = (props: Props) => (\n\t<StyledText\n\t\t{...props}\n\t\talpha={0.87}\n\t\tfamily=\"regular\"\n\t\tstyle={[styles.text, props.style]}\n\t/>\n);\n\nexport default Paragraph;\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\tfontSize: 14,\n\t\tlineHeight: 20,\n\t\tmarginVertical: 2,\n\t\tletterSpacing: 0.25,\n\t},\n});\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAAsC;AAAA;AAAA;AAAA;AAMtC;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,MAAMA,SAAS,GAAIC,KAAY,iBAC9B,oBAAC,mBAAU,eACNA,KAAK;EACT,KAAK,EAAE,IAAK;EACZ,MAAM,EAAC,SAAS;EAChB,KAAK,EAAE,CAACC,MAAM,CAACC,IAAI,EAAEF,KAAK,CAACG,KAAK;AAAE,GAEnC;AAAC,eAEaJ,SAAS;AAAA;AAExB,MAAME,MAAM,GAAGG,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,QAAQ,EAAE,EAAE;IACZC,UAAU,EAAE,EAAE;IACdC,cAAc,EAAE,CAAC;IACjBC,aAAa,EAAE;EAChB;AACD,CAAC,CAAC"}