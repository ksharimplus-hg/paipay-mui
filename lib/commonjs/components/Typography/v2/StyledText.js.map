{"version":3,"names":["StyledText","alpha","family","style","theme","themeOverrides","rest","useInternalTheme","textColor","color","isV3","colors","onSurface","text","rgb","string","writingDirection","I18nManager","getConstants","isRTL","styles","fonts","StyleSheet","create","textAlign"],"sources":["StyledText.tsx"],"sourcesContent":["import * as React from 'react';\nimport { I18nManager, StyleProp, StyleSheet, TextStyle } from 'react-native';\n\nimport color from 'color';\nimport type { ThemeProp } from 'src/types';\n\nimport Text from './Text';\nimport { useInternalTheme } from '../../../core/theming';\n\ntype Props = React.ComponentProps<typeof Text> & {\n\talpha?: number;\n\tfamily: 'regular' | 'medium' | 'light' | 'thin';\n\tstyle?: StyleProp<TextStyle>;\n\ttheme?: ThemeProp;\n};\n\nconst StyledText = ({\n\talpha = 1,\n\tfamily,\n\tstyle,\n\ttheme: themeOverrides,\n\t...rest\n}: Props) => {\n\tconst theme = useInternalTheme(themeOverrides);\n\n\tconst textColor = color(\n\t\ttheme.isV3 ? theme.colors.onSurface : theme.colors?.text\n\t)\n\t\t.alpha(alpha)\n\t\t.rgb()\n\t\t.string();\n\tconst writingDirection = I18nManager.getConstants().isRTL ? 'rtl' : 'ltr';\n\n\treturn (\n\t\t<Text\n\t\t\t{...rest}\n\t\t\tstyle={[\n\t\t\t\tstyles.text,\n\t\t\t\t{\n\t\t\t\t\tcolor: textColor,\n\t\t\t\t\t...(!theme.isV3 && theme.fonts?.[family]),\n\t\t\t\t\twritingDirection,\n\t\t\t\t},\n\t\t\t\tstyle,\n\t\t\t]}\n\t\t/>\n\t);\n};\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\ttextAlign: 'left',\n\t},\n});\n\nexport default StyledText;\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAGA;AACA;AAAyD;AAAA;AAAA;AAAA;AASzD,MAAMA,UAAU,GAAG,QAMN;EAAA;EAAA,IANO;IACnBC,KAAK,GAAG,CAAC;IACTC,MAAM;IACNC,KAAK;IACLC,KAAK,EAAEC,cAAc;IACrB,GAAGC;EACG,CAAC;EACP,MAAMF,KAAK,GAAG,IAAAG,yBAAgB,EAACF,cAAc,CAAC;EAE9C,MAAMG,SAAS,GAAG,IAAAC,cAAK,EACtBL,KAAK,CAACM,IAAI,GAAGN,KAAK,CAACO,MAAM,CAACC,SAAS,oBAAGR,KAAK,CAACO,MAAM,kDAAZ,cAAcE,IAAI,CACxD,CACCZ,KAAK,CAACA,KAAK,CAAC,CACZa,GAAG,EAAE,CACLC,MAAM,EAAE;EACV,MAAMC,gBAAgB,GAAGC,wBAAW,CAACC,YAAY,EAAE,CAACC,KAAK,GAAG,KAAK,GAAG,KAAK;EAEzE,oBACC,oBAAC,aAAI,eACAb,IAAI;IACR,KAAK,EAAE,CACNc,MAAM,CAACP,IAAI,EACX;MACCJ,KAAK,EAAED,SAAS;MAChB,IAAI,CAACJ,KAAK,CAACM,IAAI,qBAAIN,KAAK,CAACiB,KAAK,iDAAX,aAAcnB,MAAM,CAAC,EAAC;MACzCc;IACD,CAAC,EACDb,KAAK;EACJ,GACD;AAEJ,CAAC;AAED,MAAMiB,MAAM,GAAGE,uBAAU,CAACC,MAAM,CAAC;EAChCV,IAAI,EAAE;IACLW,SAAS,EAAE;EACZ;AACD,CAAC,CAAC;AAAC,eAEYxB,UAAU;AAAA"}