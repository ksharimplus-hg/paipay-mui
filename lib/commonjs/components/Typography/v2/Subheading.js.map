{"version":3,"names":["Subheading","props","styles","text","style","StyleSheet","create","fontSize","lineHeight","marginVertical","letterSpacing"],"sources":["Subheading.tsx"],"sourcesContent":["import * as React from 'react';\nimport { Text, TextStyle, StyleSheet, StyleProp } from 'react-native';\n\nimport StyledText from './StyledText';\n\nexport type Props = React.ComponentProps<typeof Text> & {\n\tstyle?: StyleProp<TextStyle>;\n\tchildren: React.ReactNode;\n};\n\n// @component-group Typography\n\n/**\n * Typography component for showing a subheading.\n *\n * <div class=\"screenshots\">\n *   <img src=\"screenshots/subheading.png\" />\n * </div>\n *\n * ## Usage\n * ```js\n * import * as React from 'react';\n * import { Subheading } from 'react-native-paper';\n *\n * const MyComponent = () => (\n *   <Subheading>Subheading</Subheading>\n * );\n *\n * export default MyComponent;\n * ```\n */\nconst Subheading = (props: Props) => (\n\t<StyledText\n\t\t{...props}\n\t\talpha={0.87}\n\t\tfamily=\"regular\"\n\t\tstyle={[styles.text, props.style]}\n\t/>\n);\n\nexport default Subheading;\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\tfontSize: 16,\n\t\tlineHeight: 24,\n\t\tmarginVertical: 2,\n\t\tletterSpacing: 0.5,\n\t},\n});\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAAsC;AAAA;AAAA;AAAA;AAOtC;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,MAAMA,UAAU,GAAIC,KAAY,iBAC/B,oBAAC,mBAAU,eACNA,KAAK;EACT,KAAK,EAAE,IAAK;EACZ,MAAM,EAAC,SAAS;EAChB,KAAK,EAAE,CAACC,MAAM,CAACC,IAAI,EAAEF,KAAK,CAACG,KAAK;AAAE,GAEnC;AAAC,eAEaJ,UAAU;AAAA;AAEzB,MAAME,MAAM,GAAGG,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,QAAQ,EAAE,EAAE;IACZC,UAAU,EAAE,EAAE;IACdC,cAAc,EAAE,CAAC;IACjBC,aAAa,EAAE;EAChB;AACD,CAAC,CAAC"}