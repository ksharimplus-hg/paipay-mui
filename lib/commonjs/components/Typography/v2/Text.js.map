{"version":3,"names":["Text","ref","style","theme","overrideTheme","rest","root","React","useRef","useInternalTheme","useImperativeHandle","setNativeProps","args","current","isV3","fonts","regular","color","colors","onSurface","text","styles","StyleSheet","create","textAlign","forwardRef"],"sources":["Text.tsx"],"sourcesContent":["import * as React from 'react';\nimport {\n\tStyleProp,\n\tStyleSheet,\n\tText as NativeText,\n\tTextStyle,\n} from 'react-native';\n\nimport type { MD2Theme } from 'src/types';\n\nimport { useInternalTheme } from '../../../core/theming';\n\ntype Props = React.ComponentProps<typeof NativeText> & {\n\tstyle?: StyleProp<TextStyle>;\n\t/**\n\t * @optional\n\t */\n\ttheme?: MD2Theme;\n};\n\n// @component-group Typography\n\n/**\n * Text component which follows styles from the theme.\n *\n * @extends Text props https://reactnative.dev/docs/text#props\n */\nconst Text: React.ForwardRefRenderFunction<{}, Props> = (\n\t{ style, theme: overrideTheme, ...rest }: Props,\n\tref\n) => {\n\tconst root = React.useRef<NativeText | null>(null);\n\tconst theme = useInternalTheme(overrideTheme);\n\n\tReact.useImperativeHandle(ref, () => ({\n\t\tsetNativeProps: (args: Object) => root.current?.setNativeProps(args),\n\t}));\n\n\treturn (\n\t\t<NativeText\n\t\t\t{...rest}\n\t\t\tref={root}\n\t\t\tstyle={[\n\t\t\t\t{\n\t\t\t\t\t...(!theme.isV3 && theme.fonts?.regular),\n\t\t\t\t\tcolor: theme.isV3\n\t\t\t\t\t\t? theme.colors?.onSurface\n\t\t\t\t\t\t: theme.colors.text,\n\t\t\t\t},\n\t\t\t\tstyles.text,\n\t\t\t\tstyle,\n\t\t\t]}\n\t\t/>\n\t);\n};\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\ttextAlign: 'left',\n\t},\n});\n\nexport default React.forwardRef(Text);\n"],"mappings":";;;;;;AAAA;AACA;AASA;AAAyD;AAAA;AAAA;AAUzD;;AAEA;AACA;AACA;AACA;AACA;AACA,MAAMA,IAA+C,GAAG,OAEvDC,GAAG,KACC;EAAA;EAAA,IAFJ;IAAEC,KAAK;IAAEC,KAAK,EAAEC,aAAa;IAAE,GAAGC;EAAY,CAAC;EAG/C,MAAMC,IAAI,GAAGC,KAAK,CAACC,MAAM,CAAoB,IAAI,CAAC;EAClD,MAAML,KAAK,GAAG,IAAAM,yBAAgB,EAACL,aAAa,CAAC;EAE7CG,KAAK,CAACG,mBAAmB,CAACT,GAAG,EAAE,OAAO;IACrCU,cAAc,EAAGC,IAAY;MAAA;MAAA,wBAAKN,IAAI,CAACO,OAAO,kDAAZ,cAAcF,cAAc,CAACC,IAAI,CAAC;IAAA;EACrE,CAAC,CAAC,CAAC;EAEH,oBACC,oBAAC,iBAAU,eACNP,IAAI;IACR,GAAG,EAAEC,IAAK;IACV,KAAK,EAAE,CACN;MACC,IAAI,CAACH,KAAK,CAACW,IAAI,qBAAIX,KAAK,CAACY,KAAK,iDAAX,aAAaC,OAAO,EAAC;MACxCC,KAAK,EAAEd,KAAK,CAACW,IAAI,oBACdX,KAAK,CAACe,MAAM,kDAAZ,cAAcC,SAAS,GACvBhB,KAAK,CAACe,MAAM,CAACE;IACjB,CAAC,EACDC,MAAM,CAACD,IAAI,EACXlB,KAAK;EACJ,GACD;AAEJ,CAAC;AAED,MAAMmB,MAAM,GAAGC,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,SAAS,EAAE;EACZ;AACD,CAAC,CAAC;AAAC,4BAEYjB,KAAK,CAACkB,UAAU,CAACzB,IAAI,CAAC;AAAA"}