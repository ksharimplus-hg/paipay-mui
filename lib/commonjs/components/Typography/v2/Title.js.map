{"version":3,"names":["Title","props","styles","text","style","StyleSheet","create","fontSize","lineHeight","marginVertical","letterSpacing"],"sources":["Title.tsx"],"sourcesContent":["import * as React from 'react';\nimport { Text, StyleSheet } from 'react-native';\n\nimport StyledText from './StyledText';\n\nexport type Props = React.ComponentProps<typeof Text> & {\n\tchildren: React.ReactNode;\n};\n\n// @component-group Typography\n\n/**\n * Typography component for showing a title.\n *\n * <div class=\"screenshots\">\n *   <img src=\"screenshots/title.png\" />\n * </div>\n *\n * ## Usage\n * ```js\n * import * as React from 'react';\n * import { Title } from 'react-native-paper';\n *\n * const MyComponent = () => (\n *   <Title>Title</Title>\n * );\n *\n * export default MyComponent;\n * ```\n */\nconst Title = (props: Props) => (\n\t<StyledText\n\t\t{...props}\n\t\talpha={0.87}\n\t\tfamily=\"medium\"\n\t\tstyle={[styles.text, props.style]}\n\t/>\n);\n\nexport default Title;\n\nconst styles = StyleSheet.create({\n\ttext: {\n\t\tfontSize: 20,\n\t\tlineHeight: 30,\n\t\tmarginVertical: 2,\n\t\tletterSpacing: 0.15,\n\t},\n});\n"],"mappings":";;;;;;AAAA;AACA;AAEA;AAAsC;AAAA;AAAA;AAAA;AAMtC;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,MAAMA,KAAK,GAAIC,KAAY,iBAC1B,oBAAC,mBAAU,eACNA,KAAK;EACT,KAAK,EAAE,IAAK;EACZ,MAAM,EAAC,QAAQ;EACf,KAAK,EAAE,CAACC,MAAM,CAACC,IAAI,EAAEF,KAAK,CAACG,KAAK;AAAE,GAEnC;AAAC,eAEaJ,KAAK;AAAA;AAEpB,MAAME,MAAM,GAAGG,uBAAU,CAACC,MAAM,CAAC;EAChCH,IAAI,EAAE;IACLI,QAAQ,EAAE,EAAE;IACZC,UAAU,EAAE,EAAE;IACdC,cAAc,EAAE,CAAC;IACjBC,aAAa,EAAE;EAChB;AACD,CAAC,CAAC"}