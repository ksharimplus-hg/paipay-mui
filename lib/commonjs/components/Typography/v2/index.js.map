{"version":3,"names":[],"sources":["index.ts"],"sourcesContent":["export { default as Caption } from './Caption';\r\nexport { default as Headline } from './Headline';\r\nexport { default as Paragraph } from './Paragraph';\r\nexport { default as Subheading } from './Subheading';\r\nexport { default as Title } from './Title';\r\n"],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA;AACA;AACA;AACA;AACA;AAA2C"}