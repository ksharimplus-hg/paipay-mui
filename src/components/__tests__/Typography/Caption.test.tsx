import * as React from 'react';
import { StyleSheet } from 'react-native';

import { render } from '@testing-library/react-native';

import { red500 } from '../../../styles/themes/v2/colors';
import Caption from '../../Typography/v2/Caption';

const style = StyleSheet.create({
  caption: {
    fontSize: 20,
    color: red500,
  },
});

it('renders caption with children as content', () => {
  const tree = render(<Caption>Caption content</Caption>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders caption applying style', () => {
  const tree = render(
    <Caption style={style.caption}>Big and red caption</Caption>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
