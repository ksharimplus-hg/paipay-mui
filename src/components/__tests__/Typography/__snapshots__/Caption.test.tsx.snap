// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders caption applying style 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(28, 27, 31, 1)",
      },
      {
        "textAlign": "left",
      },
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 0.54)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          {
            "color": "#f44336",
            "fontSize": 20,
          },
        ],
      ],
    ]
  }
>
  Big and red caption
</Text>
`;

exports[`renders caption with children as content 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(28, 27, 31, 1)",
      },
      {
        "textAlign": "left",
      },
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 0.54)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  Caption content
</Text>
`;
